featuretools/primitives/standard/transform/time_series/rolling_outlier_count.py

```diff
@@ -0,0 +1,120 @@
+        This means that aliases such as `M` or `W` cannot be used, as they can indicate different
+        numbers of days. ('M', because different months are different numbers of days;
+        'W' because week will indicate a certain day of the week, like W-Wed, so that will
+        indicate a different number of days depending on the anchoring date.)
+
+    Note:
+        When using an offset alias to define `gap`, an offset alias must also be used to define `window_length`.
+        This limitation does not exist when using an offset alias to define `window_length`. In fact,
+        if the data has a uniform sampling frequency, it is preferable to use a numeric `gap` as it is more
+        efficient.
+
+    Examples:
+        >>> import pandas as pd
+        >>> rolling_outlier_count = RollingOutlierCount(window_length=4)
+        >>> times = pd.date_range(start='2019-01-01', freq='1min', periods=6)
+        >>> rolling_outlier_count(times, [0, 0, 0, 0, 10, 0]).tolist()
+        [0.0, 0.0, 0.0, 0.0, 1.0, 1.0]
+
+        We can also control the gap before the rolling calculation.
+        >>> import pandas as pd
+        >>> rolling_outlier_count = RollingOutlierCount(window_length=4, gap=1)
+        >>> times = pd.date_range(start='2019-01-01', freq='1min', periods=6)
+        >>> rolling_outlier_count(times, [0, 0, 0, 0, 10, 0]).tolist()
+        [nan, 0.0, 0.0, 0.0, 0.0, 1.0]
+
+        We can also control the minimum number of periods required for the rolling calculation.
+        >>> import pandas as pd
+        >>> rolling_outlier_count = RollingOutlierCount(window_length=4, min_periods=3)
+        >>> times = pd.date_range(start='2019-01-01', freq='1min', periods=6)
+        >>> rolling_outlier_count(times,  [0, 0, 0, 0, 10, 0]).tolist()
+        [nan, nan, 0.0, 0.0, 1.0, 1.0]
+
+        We can also set the window_length and gap using offset alias strings.
+        >>> import pandas as pd
+        >>> rolling_outlier_count = RollingOutlierCount(window_length='4min', gap='1min')
+        >>> times = pd.date_range(start='2019-01-01', freq='1min', periods=6)
+        >>> rolling_outlier_count(times, [0, 0, 0, 0, 10, 0]).tolist()
+        [nan, 0.0, 0.0, 0.0, 0.0, 1.0]
+    """
+
+    name = "rolling_outlier_count"
+    input_types = [
+        ColumnSchema(logical_type=Datetime, semantic_tags={"time_index"}),
+        ColumnSchema(semantic_tags={"numeric"}),
+    ]
+    return_type = ColumnSchema(logical_type=Double, semantic_tags={"numeric"})
+
+    def __init__(self, window_length=3, gap=0, min_periods=0):
+        self.window_length = window_length
+        self.gap = gap
+        self.min_periods = min_periods
+
+    def get_outliers_count(self, numeric_series):
+        # We know the column is numeric, so use the Double logical type in case Woodwork's
+        # type inference could not infer a numeric type
+        if not len(numeric_series.dropna()):
```

Comment:
Do we want to return `np.nan` for windows that have no non-null values?  Alternatively, we could just return 0.